### PR TITLE
Add language tests for aggregate types

### PR DESCRIPTION
**Pull request by chorman0773**
This adds some tests for struct and union types, ensuring that they satisfy the rules for all structs and unions - namely that fields of structs do not overlap, fields are well-aligned, and the size of the entire.

The reference annotations used are from https://github.com/rust-lang/reference/pull/1654, though the rules tested here were FCPed in <https://github.com/rust-lang/reference/pull/1152>.

### CODE REVIEW

File: tests/ui/layout/aggregate-lang/struct-align.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#[allow(dead_code)]
+struct ReprRustStruct {
+    x: i32,
+    y: [u32; 4],
+    z: f32,
+    a: u128,
+    b: Overaligned,
+}
+
+#[cfg_attr(test, test)]
```
Comment (jieyouxu) [inline tests/ui/layout/aggregate-lang/struct-align.rs:19]: Question: why is this needed?

File: tests/ui/layout/aggregate-lang/struct-align.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+fn test_alignment_contains_all_fields() {
+    assert!(core::mem::align_of::<ReprRustStruct>() >= core::mem::align_of::<i32>());
+    assert!(core::mem::align_of::<ReprRustStruct>() >= core::mem::align_of::<[u32; 4]>());
+    assert!(core::mem::align_of::<ReprRustStruct>() >= core::mem::align_of::<f32>());
+    assert!(core::mem::align_of::<ReprRustStruct>() >= core::mem::align_of::<u128>());
+    assert!(core::mem::align_of::<ReprRustStruct>() >= core::mem::align_of::<Overaligned>());
+}
+
+#[cfg(not(test))]
```
Comment (jieyouxu) [inline tests/ui/layout/aggregate-lang/struct-align.rs:28]: Question: same here

**Review by jieyouxu — CHANGES_REQUESTED** [review]
Thanks, this mostly looks good to me, except for the `#[cfg_attr(test, test)]` and `#[cfg(not(test))]` attributes.

**Comment by jieyouxu** [comment]
r? jieyouxu

File: tests/ui/layout/aggregate-lang/struct-align.rs (hunk 1 above)
Comment (chorman0773) [reply tests/ui/layout/aggregate-lang/struct-align.rs:19]: This was copied out of the tests I wrote for https://github.com/lccc-project/rust-spec-conformance, which uses `--test` for run-pass tests.

Comment (jieyouxu) [reply tests/ui/layout/aggregate-lang/struct-align.rs:19]: Ah okay. Please remove these `--test` annotations, then we can merge these tests.

Comment (jieyouxu) [reply tests/ui/layout/aggregate-lang/struct-align.rs:19]: You can r=me after PR CI is green and removing those `--test`-related annotations.

**Comment by jieyouxu** [comment]
You can r=me after removing those `--test`-related annotations and PR CI is green.
@bors delegate+
